### DISCUSSION TITLE
sphinx.ext.autodoc' has no attribute 'startswith

### DISCUSSION THREAD

**Discussion by radi8** [sphinx-doc/sphinx#12269] [Q&A] [answered]
### Describe the bug

When trying to compile I receive an error message. Below is the printout from my terminal showing the output of the session:

```console
Rpi4:~/Programs $ sphinx-build -b latex . _build
Running Sphinx v7.2.6

Exception occurred:
  File "/usr/lib/python3.9/importlib/__init__.py", line 118, in import_module
    if name.startswith('.'):
AttributeError: module 'sphinx.ext.autodoc' has no attribute 'startswith'
The full traceback has been saved in /tmp/sphinx-err-wnfpeocc.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Rpi4:~/Programs $
```

### How to Reproduce

run from the command line: sphinx-build -b latex . _build

conf.py:

```python
import sphinx
from sphinx.ext.autodoc import setup

project = 'remoteServer'
copyright = '2024, Graeme Jury ZL2TE'
author = 'Graeme Jury ZL2TE'
release = 'remoteServer'
extensions = [sphinx.ext.autodoc]
templates_path = ['_templates']
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']

html_theme = 'alabaster'
html_static_path = ['_static']
```

index.rst:
```rest
Welcome to remoteServer's documentation!
========================================

.. toctree::
   :maxdepth: 2
   :caption: Contents:

   MremoteServer
   registerClass
```

MremoteServer.rst:
```rest
.. _MremoteServer:

MremoteServer module
====================

.. automodule:: MremoteServer
   :members:
   :undoc-members:
   :show-inheritance:
```

registerClass.rst:
```rest
.. _registerClass:

registerClass module
=====================

.. automodule:: registerClass
   :members:
   :undoc-members:
   :show-inheritance:
```

### Environment Information

```text
Platform:              linux; (Linux-6.1.21-v8+-aarch64-with-glibc2.31)
Python version:        3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110])
Python implementation: CPython
Sphinx version:        7.2.6
Docutils version:      0.20.1
Jinja2 version:        3.1.3
Pygments version:      2.17.2
```

### Sphinx extensions

No extensions but this is the first time I have used this software and am a complete newbie.

### Additional context

Rpi4:~/Programs $ latex
This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020/Debian) (preloaded format=latex)
 restricted \write18 enabled.

**Comment by radi8** [comment]
[sphinx-err-wnfpeocc.log](https://github.com/sphinx-doc/sphinx/files/14951762/sphinx-err-wnfpeocc.log)

**Comment by jayaddison** [comment, marked as answer]
> `extensions = [sphinx.ext.autodoc]`

Hi @radi8 - thanks for the bugreport.  Is this line something that appears in your `conf.py` file?  And if so, could you try updating it to make sure that the extension name is a string value: `extensions = ['sphinx.ext.autodoc']` (note the quotes around the extension module name)?

**Reply by AA-Turner** [reply]
Hi Graeme -- as James points out, the extensions need to be a list of strings, which are interpreted by Sphinx as import paths for Python modules.

As such, the ``from sphinx.ext.autodoc import setup`` line in your ``conf.py`` isn't needed -- the extension will be set up automatically.

A

**Reply by radi8** [reply]
Thank you all for steering me on the right course. I have removed the extension (I got an error message "WARNING: while setting up extension conf.py: directive 'automodule' is already registered, it will be overridden"). I'm still not out of the woods as I'm getting the following warnings now:
WARNING: autodoc: failed to import module 'MremoteServer'; the following exception was raised:
No module named 'MremoteServer'
WARNING: autodoc: failed to import module 'registerClass'; the following exception was raised:
No module named 'registerClass'
I guess this is from my not knowing how to properly set it up so off to read the manual.

Many thanks, Graeme
